MyCapytain/resolvers/capitains/local.py

```diff
@@ -0,0 +1,497 @@
+                text_metadata.citation = cites[-1]
+                self.logger.info("%s has been parsed ", text_metadata.path)
+                if not text_metadata.citation.is_set():
+                    self.logger.error("%s has no passages", text_metadata.path)
+                    return False
+                return True
+            except Exception:
+                self.logger.error(
+                    "%s does not accept parsing at some level (most probably citation) ",
+                    text_metadata.path
+                )
+                return False
+        else:
+            self.logger.error("%s is not present", text_metadata.path)
+            return False
+
+    def _dispatch(self, collection, directory):
+        """ Run the dispatcher over a textgroup.
+
+        :param collection: Collection object that needs to be dispatched
+        :param directory: Directory in which the Collection was found
+        """
+        if collection.id in self.dispatcher.collection:
+            self.dispatcher.collection[collection.id].update(collection)
+        else:
+            self.dispatcher.dispatch(collection, path=directory)
+
+        for sub_coll_id, sub_coll in collection.collections.items():
+            if sub_coll_id in self.dispatcher.collection[collection.id].collections:
+                self.dispatcher.collection[sub_coll_id].update(sub_coll)
+
+    def _dispatch_container(self, collection, directory):
+        """ Run the dispatcher over a collection within a try/except block
+
+        .. note:: This extraction allows to change the dispatch routine \
+            without having to care for the error dispatching
+
+        :param collection: Collection object that needs to be dispatched
+        :param directory: Directory in which the Collection was found
+        """
+        try:
+            self._dispatch(collection, directory)
+        except UndispatchedTextError as E:
+            self.logger.error("Error dispatching %s ", directory)
+            if self.RAISE_ON_UNDISPATCHED is True:
+                raise E
+
+    def _clean_invalids(self, invalids):
+        """ Optionally remove texts that were found to be invalid
+
+        :param invalids: List of text identifiers
+        :type invalids: [CapitainsReadableMetadata]
+        """
+        pass
+
+    def parse(self, resource):
+        """ Parse a list of directories and reads it into a collection
+
+        :param resource: List of folders
+        :return: An inventory resource and a list of CapitainsReadableMetadata metadata-objects
+        """
+        id_to_coll = dict()
+        members = dict()
+        parents = dict()
+        invalids = []
+
+        for folder in resource:
+            metadata_files = glob("{base_folder}/data/**/__capitains__.xml".format(base_folder=folder), recursive=True)
+            for metadata_file in metadata_files:
+                collection, children, rel_dir = self._parse_collection_wrapper(metadata_file)
```

Comment:
`collection, children, rel_dir, parents_relationships = self._parse....`

Reply:
?

Reply:
Probably `parents_relationships` and `child_relationships` since both can exist in the metadata. And since this is coming from parsing the `__capitains__.xml` files, that is what should be represented here. Deeper or different relationships should be discovered by the resolver, I think.

Reply:
Also...next PR.